src/ex1/script.js

```diff
@@ -0,0 +1,213 @@
+        document.getElementById(i + 1).remove();
+        last_page--;
+      }
+    }
+  }
+  DisplayPage(current_page);
+}
+
+function addTaskToHTML(task) {
+  taskToAdd.value = "";
+  document.querySelector("#tasks").innerHTML += `
+            <div class="task">
+                <span id="taskname" style="width:600px;">
+                    ${task}
+                </span>
+                <button class="delete">
+                    <i class="far fa-trash-alt"></i>
+                </button>
+            </div>
+    `;
+  let current_tasks = document.querySelectorAll(".delete");
+  for (var i = 0; i < current_tasks.length; i++) {
+    current_tasks[i].onclick = function () {
+      let index = page_to_tasks_map
+        .get(current_page)
+        .indexOf(this.parentNode.innerText);
+      tmp_arr = page_to_tasks_map.get(current_page);
+      tmp_arr.splice(index, 1);
+      page_to_tasks_map.set(current_page, tmp_arr);
+      index = chronological_arr.indexOf(this.parentNode.innerText);
+      chronological_arr.splice(index, 1);
+      this.parentNode.remove();
+      if (
+        current_page === 1 &&
+        page_to_tasks_map.get(current_page).length === 0
+      )
+        return;
+      updatePages();
+    };
+  }
+
+  let current_tasks_2 = document.querySelectorAll('[id=taskname]');
+  for (let i = 0; i < current_tasks_2.length; i++) {
+    current_tasks_2[i].onclick = function () {
+      alert(current_tasks_2[i].parentNode.innerText);
+    };
+  }
+}
+
+function onAddBtnClick() {
+  let task = taskToAdd.value;
+  if (page_to_tasks_map.size) {
+    for (let i = 1; i < page_to_tasks_map.size + 1; i++) {
+      let arr_tasks = page_to_tasks_map.get(i);
+      if (arr_tasks.includes(task)) {
+        alert("task already in, find it at page: " + i);
+        taskToAdd.value = "";
+        return;
+      }
+    }
+  }
+  chronological_arr.push(task);
+  if (page_to_tasks_map.get(current_page).length === 5) {
+    if (page_to_tasks_map.get(last_page).length === 5) {
+      tasks = [];
+      tasks.push(task);
+      last_page++;
+      page_to_tasks_map.set(last_page, tasks);
+      document.querySelector("#tasks").innerHTML = ``;
+
+      let btn = PaginationButton(page_to_tasks_map.size);
+      pagination_element.appendChild(btn);
+
+      DisplayPage(last_page);
+    } else {
+      page_to_tasks_map.set(last_page, [
+        ...page_to_tasks_map.get(last_page),
+        task,
+      ]);
+      DisplayPage(last_page);
+    }
+  } else {
+    page_to_tasks_map.set(current_page, [
+      ...page_to_tasks_map.get(current_page),
+      task,
+    ]);
+    DisplayPage(last_page);
+  }
+}
+
+function DisplayPage(page) {
+  let page_btn = document.getElementById(current_page);
+  if (current_page == page) page_btn.classList.add("active");
+  current_page = page;
+  elements = page_to_tasks_map.get(current_page);
+  document.querySelector("#tasks").innerHTML = ``;
+  for (let i = 0; i < elements.length; i++) {
```

Comment:
try to avoid using `for i` and use `for const of` - much easier to handle.
see example: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of